### PR TITLE
Add database

### PR DESCRIPTION
**Pull request by Jaunius343**

### CODE REVIEW

**Comment by Jaunius343** [comment]
I need help, I managed to break your subscription feature. I'll comment the lines which concern me

File: PSI Food waste/Pages/Forms/RestaurantList.cshtml [hunk 1]
```diff
@@ -44,6 +44,7 @@
             {
                 if (@Model._registerRepository.CurrentUser.SubscribedRestaurants.Contains(restaurant))
```
Comment (Jaunius343) [inline PSI Food waste/Pages/Forms/RestaurantList.cshtml:45]: Here, even if restaurant is in the list, the subscribe button still appears. On top of that, in the page model, it adds restaurant duplicates, as if the check "contains" is no longer working

Comment (Pachtakoras) [reply PSI Food waste/Pages/Forms/RestaurantList.cshtml:45]: Not really sure, I first thought it was that when a restaurants gets added to the data base the primary key gets assigned to it, so the restaurant gets added to the current user list without a id. But it turns out wrong

**Comment by Pachtakoras** [comment]
Yes I was about to write that prices are not shown correctly.

File: PSI Food waste/Pages/Forms/RestaurantList.cshtml (hunk 1 above)
Comment (Jaunius343) [reply PSI Food waste/Pages/Forms/RestaurantList.cshtml:45]: I have done some testing and it seems that the compare is broken. Even if restaurants are the same, contains function still returns false. "Equals" returns same results, always false, even if objects have the same values. The only thing that worked for me right now is matching restaurant IDs. I might make it as a temporary solution

Comment (Jaunius343) [reply PSI Food waste/Pages/Forms/RestaurantList.cshtml:45]: ok, so it was checking by reference, no clue how and why it was working before. I've overridden the Equals functions and now it works as expected.

**Pull request merged**